### PR TITLE
Span integration test

### LINKED ISSUE #54: Create integration test(s) for the Trace/Span API

### PR DESCRIPTION
**Pull request by XiXiaPdx**
resolves #54

### CODE REVIEW

File: telemetry_examples/src/main/java/com/newrelic/telemetry/examples/SpanExample.java [hunk 1]
```diff
@@ -49,6 +50,7 @@ public static void main(String[] args) throws Exception {
             .httpPoster(new OkHttpPoster(Duration.ofSeconds(5)))
             .attributesJson(new AttributesGson(new Gson()))
             .enableAuditLogging()
+            .uriOverride(URI.create("https://staging-trace-api.newrelic.com"))
```
Comment (breedx-nr) [inline telemetry_examples/src/main/java/com/newrelic/telemetry/examples/SpanExample.java]: I'm thinking that we don't want our examples pointing at staging, we should just omit this for clarity and brevity.

File: integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                .withMethod("POST")
+                .withPath("/trace/v1")
+                .withBody(
+                    json(
+                        new SpanPayload[] {expectedPayload},
+                        MediaType.JSON_UTF_8,
+                        MatchType.ONLY_MATCHING_FIELDS))
+                .withHeader("User-Agent", "NewRelic-Java-TelemetrySDK/.*")
+                .withHeader("Content-Type", "application/json; charset=utf-8")
+                .withHeader("Content-Length", ".*"))
```
Comment (breedx-nr) [inline integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java]: Is this helpful, or can we omit the wildcard length check?

File: integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            ImmutableMap.of("trace.id", "123456", "attributes", singletonMap("key1", "val1")),
+            singletonList(
+                ImmutableMap.<String, Object>builder()
+                    .put("id", "6f377f46-3f10-4e8e-8390-7717044cdbe9")
+                    .put("trace.id", "123456")
+                    .put("timestamp", 55555)
+                    .put(
+                        "attributes",
+                        ImmutableMap.of(
+                            "duration.ms",
```
Comment (breedx-nr) [inline integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java]: nitpick: might be nice to see if these pairs will fit on the same line (reads more like mappings that way).

File: integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java (hunk 2 above)
Comment (jkwatson) [reply integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java]: I think this is a good thing to check...it's well-behaved to provide a content length header.

File: telemetry_examples/src/main/java/com/newrelic/telemetry/examples/SpanExample.java (hunk 1 above)
Comment (jkwatson) [reply telemetry_examples/src/main/java/com/newrelic/telemetry/examples/SpanExample.java]: 👍

File: integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java (hunk 3 above)
Comment (XiXiaPdx) [reply integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java]: fixed

File: telemetry_examples/src/main/java/com/newrelic/telemetry/examples/SpanExample.java (hunk 1 above)
Comment (XiXiaPdx) [reply telemetry_examples/src/main/java/com/newrelic/telemetry/examples/SpanExample.java]: removed

**Pull request merged**